packages/map/src/popup/index.jsx

```diff
@@ -0,0 +1,58 @@
+import { useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+import { Popup as MapboxPopup } from 'mapbox-gl';
+
+import { useMarkerEffect } from '../marker/hooks';
+
+
+const Popup = ({
+  className,
+  children,
+  defaultOpen,
+  offsetX,
+  offsetY,
+}) => {
+  const nodeRef = useRef();
+  const [, forceRender] = useState();
+
+  useMarkerEffect((marker) => {
+    const node = document.createElement('div');
+    node.className = className;
+
+    const popup = new MapboxPopup({ offset: [offsetX, offsetY] }).setDOMContent(node);
+    marker.setPopup(popup);
+    if (defaultOpen) marker.togglePopup();
+
+    nodeRef.current = node;
+    forceRender({});
+
+    return () => {
+      popup.remove();
+      nodeRef.current = null;
+      forceRender({});
+    };
+  }, [children, offsetX, offsetY, className]);
+
+  if (nodeRef.current) {
+    // Using portal since mapbox moves DOM nodes to body
+    return createPortal(children, nodeRef.current);
+  }
+
+  return null;
+};
+
+Popup.defaultProps = {
+  offsetX: 0,
+  offsetY: 0,
+};
+
+Popup.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node,
+  defaultOpen: PropTypes.bool,
```

Comment:
Possible issue: Pretty similar to nebenan-forms Inputs, we are building here an uncontrolled component. I think it's way easier to have controlled components and if you really don't want to handle their state changes, you build a wrapper for it. 

I see myself needing a state controlled popup in the future

Reply:
On second thought, this will require handling misclicks as well and we will need to write a wrapper to hold a state because in most cases we don't want to control it.

There is a pretty simple fix if you want to control popup state -- use key prop